### DISCUSSION TITLE
Create NiceGUI command line tool

### DISCUSSION THREAD

**Discussion by qingant** [zauberzeug/nicegui#1462] [Ideas / Feature Requests]
I like the simplicity of NiceGUI. but there would definitely be some magic that I would like to have even though it will add a little bit complexity:
1. currently, if we wrote code in the entry module without router decorator, it's shared application wide which is almost never wanted --- even though understandable, it should be wrapped into `ui.route('/')` by default.
2. Some options would rather be placed on command line (native, on_air, etc) , currently, the developer will need to build the cli on their entry module
3. Better frontend tooling, it could be a good thing if we can have some support on frontend tooling to enable developer to write NiceGUI vue component easier. Having a command line is a way to standardize the tooling (you don't want to force every component developer to reinvent the tooling wheels ).

These all kind of require a command line tool.

**Comment by rodja** [comment]
Thanks for starting this discussion @qingant. To write the truth I'm not sure if NiceGUI really should have a CIL.

> it should be wrapped into ui.route('/') by default.

I think that would be quite complicated because there could be any amount of non-ui stuff going on. Like accessing an USB camera, initializing a database etc. This is one of the main reasons why [Streamlit is so broken](https://github.com/zauberzeug/nicegui/discussions/21). I would favour something like #1065.

> Some options would rather be placed on command line (native, on_air, etc) 

I do not think these options are often required to change. And if they need to be configured they may be better placed in an .env file. CILs are not a great way to configure a deployment.

> it could be a good thing if we can have some support on frontend tooling

Maybe. Depends on the outcome of #1454 in my opinion.
